src/main/java/dao/WriterImpl.java

```diff
@@ -0,0 +1,30 @@
+package dao;
+
+import java.io.BufferedWriter;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.Map;
+import service.ReportImpl;
+
+public class WriterImpl implements Writer {
+    private static final String FILE_PATH = "src/main/reports/report.csv";
+    private static final Map<String, Integer> reportMap = new ReportImpl().report();
+
+    @Override
+    public void reportWriter() {
```

Comment:
```suggestion
    public void reportWriter(String fileName) {
```
it's better to dynamically pass fileName instead of hardcoding it